src/GraphWebhooks-Core/Controllers/SubscriptionController.cs

```diff
@@ -10,53 +10,59 @@
 using Microsoft.Graph;
 using GraphWebhooks_Core.Helpers;
 using Microsoft.Extensions.Options;
+using Microsoft.Identity.Web.Client;
+using GraphWebhooks_Core.Infrastructure;
+using GraphWebhooks_Core.Helpers.Interfaces;
 
 namespace GraphWebhooks_Core.Controllers
 {
     [Authorize]
-    [ValidateAntiForgeryToken]
     public class SubscriptionController : Controller
     {
-        private readonly ISDKHelper sdkHelper;
         private readonly ISubscriptionStore subscriptionStore;
         private readonly AppSettings appSettings;
-        
-        public SubscriptionController(ISDKHelper sdkHelper,
-                                      ISubscriptionStore subscriptionStore,
-                                      IOptions<AppSettings> optionsAccessor)
-        {
-            this.sdkHelper = sdkHelper;
+        readonly ITokenAcquisition tokenAcquisition;
+
+        public SubscriptionController(ISubscriptionStore subscriptionStore,
+                                      IOptions<AppSettings> optionsAccessor,
+                                      ITokenAcquisition tokenAcquisition)
+        {            
             this.subscriptionStore = subscriptionStore;
             appSettings = optionsAccessor.Value;
+            this.tokenAcquisition = tokenAcquisition;
         }
 
         // Create a subscription.
+        [MsalUiRequiredExceptionFilter(Scopes = new[] { Infrastructure.Constants.ScopeMailRead })]
         public async Task<IActionResult> Create()
         {
             string userId = User.FindFirst("http://schemas.microsoft.com/identity/claims/objectidentifier")?.Value;
             string tenantId = User.FindFirst("http://schemas.microsoft.com/identity/claims/tenantid")?.Value;
             string clientState = Guid.NewGuid().ToString();
 
+            // Fetch the access token
+            string accessToken = await tokenAcquisition.GetAccessTokenOnBehalfOfUser(
+                    HttpContext, new[] { Infrastructure.Constants.ScopeMailRead });
+
             Subscription newSubscription = new Subscription();
             try
             {
-
-                // Initialize the GraphServiceClient. 
-                // This sample passes in the tenant ID to use as a cache key.
-                GraphServiceClient graphClient = sdkHelper.GetAuthenticatedClient(tenantId);
+                // Initialize the GraphServiceClient.                
+                var graphClient = GraphServiceClientFactory.GetAuthenticatedGraphClient(accessToken);
```

Comment:
Same remark: if the graph client needs to get the token several times during more than the lifetime of the token, rather use a delegate

Reply:
Done the same for all the calls to `GetAuthenticatedClient`